pkg/udn/primary_network.go

```diff
@@ -0,0 +1,39 @@
+package udn
+
+import (
+	"context"
+	"fmt"
+
+	v1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kubevirt/ipam-extensions/pkg/config"
+)
+
+func FindPrimaryNetwork(ctx context.Context,
+	cli client.Client,
+	namespace string) (*v1.NetworkAttachmentDefinition, error) {
+	nadList := v1.NetworkAttachmentDefinitionList{}
+	if err := cli.List(ctx, &nadList, client.InNamespace(namespace)); err != nil {
+		return nil, fmt.Errorf("failed listing nads for pod namespace")
+	}
+
+	var primaryNetworkNAD *v1.NetworkAttachmentDefinition
+	for _, nad := range nadList.Items {
+		netConfig, err := config.NewConfig(nad.Spec.Config)
+		if err != nil {
+			return nil, fmt.Errorf("failed to extract the relevant NAD information")
+		}
+		if netConfig.Role == config.NetworkRolePrimary {
+			if primaryNetworkNAD != nil {
+				return nil, fmt.Errorf("multiple primary network '%s/%s' and '%s/%s' found, "+"only one primary "+
+					"network should be configured in a given namespace",
+					nad.Namespace, nad.Name, primaryNetworkNAD.Namespace, primaryNetworkNAD.Name)
```

Comment:
should we even care about this ? Why don't we just return the first, and let OVN-K explode in all its glory with this error ?

Reply:
Let's do that.